mobile/library/cc/engine_builder.cc

```diff
@@ -754,6 +766,20 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     (*base_cluster->mutable_typed_extension_protocol_options())
         ["envoy.extensions.upstreams.http.v3.HttpProtocolOptions"]
             .PackFrom(alpn_options);
+
+    // Set the upstream connections socket receive buffer size. The operating system defaults are
+    // usually too small for QUIC.
+    // TODO(32304): We can remove this once core Envoy has better receive buffer defaults.
```

Comment:
I think we should explicitly set this in EM regardless of the default value in core Envoy. We know 1MB is sufficient for mobile but not necessarily for server upstream connections.

Reply:
fair point!

Reply:
oops sorry just saw this.  do we plan on making the core Envoy change configured via `bind_options` just like we do here? in that case, we probably don't need this TODO, and I'll send a follow up PR to remove it

Reply:
https://github.com/envoyproxy/envoy/pull/32337